scripts/launch.py

```diff
@@ -0,0 +1,339 @@
+        return params
+
+    def get_config(self):
+        members = [{'procParams': self.proc_params()}]
+        if not self.single:
+            members.extend([
+                {'procParams': self.proc_params()},
+                {#'rsParams': {'arbiterOnly': True},
+                 'procParams': self.proc_params()}
+            ])
+        return {'members': members}
+
+    def _init_from_response(self, response):
+        self.id = response['id']
+        self.uri = response.get('mongodb_auth_uri', response['mongodb_uri'])
+        for member in response['members']:
+            m = Server(member['server_id'], member['host'])
+            self.members.append(m)
+            if member['state'] == 1:
+                self.primary = m
+            elif member['state'] == 2:
+                self.secondary = m
+        return self
+
+    def start(self):
+        # We never need to restart a replica set, only start new ones.
+        return self._init_from_response(self._make_post_request())
+
+    def restart_primary(self):
+        self.primary.stop(destroy=False)
+        time.sleep(5)
+        self.primary.start()
+        time.sleep(1)
+        self._init_from_response(self._make_get_request())
+        print('New primary: %s' % self.primary.uri)
+
+
+class ReplicaSetSingle(ReplicaSet):
+
+    def get_config(self):
+        return {
+            'members': [
+                {'procParams': self.proc_params()}
+            ]
+        }
+
+
+class ShardedCluster(MCTestObject):
+
+    _resource = 'sharded_clusters'
+    _shard_type = ReplicaSet
+
+    def __init__(self, **kwargs):
+        self.id = None
+        self.uri = None
+        self.shards = []
+        self._proc_params = kwargs
+
+    def get_config(self):
+        return {
+            # 'configsvrs': [{'members': [DEFAULT_OPTIONS.copy()]}],
+            'routers': [self.proc_params(), self.proc_params()],
+            'shards': [
+                {'id': 'demo-set-0', 'shardParams':
+                    self._shard_type().get_config()},
+                # {'id': 'demo-set-1', 'shardParams':
+                #     self._shard_type().get_config()}
+            ]
+        }
+
+    def start(self):
+        # We never need to restart a sharded cluster, only start new ones.
+        response = self._make_post_request()
+        for shard in response['shards']:
+            shard_resp = requests.get(_mo_url('replica_sets', shard['_id']))
+            shard_json = shard_resp.json()
+            self.shards.append(self._shard_type()._init_from_response(shard_json))
+        self.id = response['id']
+        self.uri = response.get('mongodb_auth_uri', response['mongodb_uri'])
+        return self
+
+
+class ShardedClusterSingle(ShardedCluster):
+    _shard_type = ReplicaSetSingle
+
+
+def argv_has(string):
+    return any(string in arg for arg in sys.argv[1:])
+
+
+CERTS = None # REPLACE WITH THE PATH TO YOUR *.pem FILES IF YOU WANT SSL SUPPORT
```

Comment:
Now that this is included in the repo and shipped with MO itself I realize it's very awkward to need to edit this file for tls support. Let's add pymongo's testing certs here (ca.pem/server.pem): https://github.com/10gen/mongo-orchestration/tree/master/mongo_orchestration/lib